### ISSUE TITLE
Garden-ui app should not ask for switch network if not the active tab

### ISSUE THREAD

**Issue by Corantin** [1Hive/gardens-ui#866]
### Problem
Switch wallet network and Gardens-UI opened in another tab resulting in metamask asking us to switch network back to the one before.

### Repro
1. We change network in another chain (from gnosis to Rinkeby)
![image](https://user-images.githubusercontent.com/13890618/180627489-c130d1dd-d73f-40ec-83db-5b8c41b955d1.png)

2. Gardens ask to switch network
![image](https://user-images.githubusercontent.com/13890618/180627459-6dc36d75-483f-42a9-9570-de867c35e981.png)

### Suggestion
Only ask to chain network when Gardens-UI app is the active tab
> Example in Quests : https://github.com/1Hive/quests/blob/main/packages/react-app/src/contexts/wallet.context.tsx#:~:text=%20!document.hidden%20%26%26

**Comment by kamikazebr** [comment]
Yeah the #825 have the idea fix that, but its need some review to be approved.
